sdk/core/Azure.Core/api/Azure.Core.net461.cs

```diff
@@ -678,6 +678,14 @@ public abstract partial class TokenCredential
         public string[] Scopes { get { throw null; } }
         public string? TenantId { get { throw null; } }
     }
+    public partial class UserAgentValue
+    {
+        public UserAgentValue(System.Type packageType, string? applicationId = null) { }
```

Comment:
How is this actually used by library authors?  I was wondering if we could remove `applicationId` here so people don't have to track it and have the telemetry policy pass it in instead.  I don't think we should allow individual libraries to ignore the applicationId which could happen if it has to optionally be passed here.

Reply:
example usage is in the description here https://github.com/Azure/azure-sdk-for-net/pull/27079

The main concern there is that we want the string to be constructed once and cached. Client authors should pass this along from the clientOptions if it's available. But if we don't apply it here, we are forced to do it at request time in the TelemetryPolicy. The alternative, which I had at one point during iteration, was to accept ClientOptions here, but it still relies on the client author to pass the one created by the customer.